### PR TITLE
Support get search attributes inside workflow

### PR DESCRIPTION
**Pull request by vancexu**
This PR support get search attributes in workflow using GetInfo. 
Also support related test in testsuite.

### CODE REVIEW

File: internal/internal_event_handlers.go [hunk 1]
```diff
@@ -297,9 +297,22 @@ func (wc *workflowEnvironmentImpl) UpsertSearchAttributes(attributes map[string]
 
 	upsertID := wc.GenerateSequenceID()
 	wc.decisionsHelper.upsertSearchAttributes(upsertID, attr)
+	wc.workflowInfo.SearchAttributes = mergeSearchAttributes(wc.workflowInfo.SearchAttributes, attr)
```
Comment (shreyassrivatsan) [inline internal/internal_event_handlers.go]: I dont like the idea of overriding the search attributes one off like this.. As part of processing the event we should have a cleaner way to update `WorkflowInfo`.. I'll need to think whats the best way to handle that..

File: internal/internal_event_handlers_test.go [hunk 2]
```diff
@@ -164,10 +165,44 @@ func Test_ValidateAndSerializeSearchAttributes(t *testing.T) {
...
 	}
 	err := env.UpsertSearchAttributes(nil)
 	require.Error(t, err)
 
 	err = env.UpsertSearchAttributes(map[string]interface{}{"key": 1})
 	require.NoError(t, err)
 }
+
+func Test_MergeSearchAttributes(t *testing.T) {
+	var current *s.SearchAttributes
```
[the remark this answers is not in the document]
Comment (vancexu) [reply internal/internal_event_handlers_test.go]: done

File: internal/internal_event_handlers.go (hunk 1 above)
Comment (vancexu) [reply internal/internal_event_handlers.go]: Ok, let me know if better option

Comment (shreyassrivatsan) [reply internal/internal_event_handlers.go]: Actually is this even the right place to have this? The user calls upsertSearchAttributes and that should cause a call to the server. The value on the client should only get updated while processing the event - not prior to the call and the server making the change.. Am I missing something?

Comment (vancexu) [reply internal/internal_event_handlers.go]: You are right. Changed to update in process event.

**Pull request merged**